Generator.Equals/Models/EqualityTypeModelTransformer.cs

```diff
@@ -0,0 +1,112 @@
+﻿using System;
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using Generator.Equals.Extensions;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+namespace Generator.Equals.Models;
+
+internal sealed class EqualityTypeModelTransformer
+{
+    private readonly GeneratorAttributeSyntaxContext _context;
+    private readonly CancellationToken _token;
+
+    public EqualityTypeModelTransformer(GeneratorAttributeSyntaxContext context, CancellationToken token)
+    {
+        _context = context;
+        _token = token;
+    }
+
+    public EqualityTypeModel Transform()
+    {
+        var attributesMetadata = AttributesMetadata.Instance;
+
+        var equatableAttributeData = _context.Attributes.SingleOrDefault();
+        if (equatableAttributeData is null || !attributesMetadata.Equatable.Equals(equatableAttributeData.AttributeClass))
+        {
+            throw new Exception("Expected exactly one EquatableAttribute.");
+        }
+
+
+        var explicitMode = equatableAttributeData.GetNamedArgumentValue("Explicit") is true;
+        var ignoreInheritedMembers = equatableAttributeData.GetNamedArgumentValue("IgnoreInheritedMembers") is true;
+
+        if (_context.TargetSymbol is not ITypeSymbol symbol)
+        {
+            throw new Exception("Expected a type symbol.");
+        }
+
+        var baseTypeName = symbol.BaseType?.ToFQF();
+
+        // Extract the symbol's fully qualified name
+        var typeName = symbol.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat);
+
+        var fullname = symbol.ToDisplayString();
```

Comment:
Use an explicit format, not just the default.

Reply:
Seems like only this works. It is only used for ``productionContext.AddSource`` as filename. Other ones lead to the generator failing because of duplicate names.

Reply:
Use a format that works for you. The default is `CSharpErrorMessageFormat`, and that is likely not a good one for using as a file name.